pkg/util/push/request.go

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package push
+
+import (
+	"context"
+	"net/http"
+	"sync"
+
+	"github.com/grafana/mimir/pkg/mimirpb"
+)
+
+type Request struct {
+	cleanups []func()
+
+	httpReq                      *http.Request
+	maxMessageSize               int
+	allowSkipLabelNameValidation bool
+	parser                       ParserFunc
+	parseRequestOnce             sync.Once
+	parsedRequest                *mimirpb.WriteRequest
+	parseError                   error
+}
+
+func (r *Request) parseRequest(ctx context.Context) {
+	bufHolder := bufferPool.Get().(*bufHolder)
+	var req mimirpb.PreallocWriteRequest
+	buf, err := r.parser(ctx, r.httpReq, r.maxMessageSize, bufHolder.buf, &req)
+	if err != nil {
+		//level.Error(logger).Log("err", err.Error())
+		//http.Error(w, err.Error(), http.StatusBadRequest)
```

Comment:
I had forgotten about this and now realizing that this is introducing a bug - these errors will likely be reported as a 5xx by the `push.handler`. I'm working on a fix...

Reply:
should be sorted in 3992f7bba6adac35cf756038fde29dd4884c4485